service/tablet_allocator.cc

```diff
@@ -432,14 +442,41 @@ class load_balancer {
         }
 
         // Call when tablet_count changes.
-        void update() {
-            avg_load = get_avg_load(tablet_count);
+        void update(uint64_t target_tablet_size) {
+            if (capacity) {
+                avg_load = *get_avg_load(tablet_count, target_tablet_size);
+            }
+        }
+
+        // Result engaged when !drained.
+        std::optional<load_type> get_avg_load(uint64_t tablets, uint64_t target_tablet_size) const {
+            if (!capacity) {
+                return std::nullopt;
+            }
+            return double(tablets) * target_tablet_size / *capacity;
         }
 
-        load_type get_avg_load(uint64_t tablets) const {
+        double tablets_per_shard(uint64_t tablets) const {
             return double(tablets) / shard_count;
         }
 
+        double tablets_per_shard() const {
+            return tablets_per_shard(tablet_count);
+        }
+
+        // Result engaged for !drained nodes.
+        std::optional<load_type> shard_load(shard_id shard, uint64_t target_tablet_size) const {
+            return shard_load(shard, shards[shard].tablet_count, target_tablet_size);
+        }
+
+        // Result engaged for !drained nodes.
+        std::optional<load_type> shard_load(shard_id shard, uint64_t shard_tablet_count, uint64_t target_tablet_size) const {
+            if (!capacity) {
```

Comment:
I guess that the compiler may very well optimize this, but we could call `capacity_per_shard` and check its result rather than double checking `capacity` at each level.

Reply:
Ok